price-server/src/exchange_price_cache.rs

```diff
@@ -30,23 +29,22 @@ impl ExchangePriceCache {
         self.inner.write().await.update_price(payload);
     }
 
-    pub async fn ask_price_of_one_sat(&self) -> Result<UsdCents, ExchangePriceCacheError> {
-        self.inner.read().await.ask_price_of_one_sat()
-    }
-
-    pub async fn bid_price_of_one_sat(&self) -> Result<UsdCents, ExchangePriceCacheError> {
-        self.inner.read().await.bid_price_of_one_sat()
-    }
-
-    pub async fn mid_price_of_one_sat(&self) -> Result<UsdCents, ExchangePriceCacheError> {
-        self.inner.read().await.mid_price_of_one_sat()
+    pub async fn latest_tick(&self) -> Result<BtcSatTick, ExchangePriceCacheError> {
+        self.inner.read().await.latest_tick()
     }
 }
 
-struct BtcSatTick {
+#[derive(Clone)]
+pub struct BtcSatTick {
     timestamp: TimeStamp,
-    ask_price_of_one_sat: UsdCents,
-    bid_price_of_one_sat: UsdCents,
+    pub ask_price_of_one_sat: UsdCents,
+    pub bid_price_of_one_sat: UsdCents,
+}
+
+impl BtcSatTick {
+    pub fn mid_price_of_one_sat(&self) -> UsdCents {
+        (self.bid_price_of_one_sat.clone() + self.ask_price_of_one_sat.clone()) / 2
```

Comment:
Too much `.clone()`ing. You need to `impl Add` on the pointer type `&UsdCents`.
Something like:
```
        impl std::ops::Add<&$name> for &$name {
            type Output = $name;

            fn add(self, rhs: &$name) -> Self::Output {
                Self {
                    inner: self.inner + res
                }
            }
        }

```

Reply:
I have a blocker implementing `Add` for the pointer type `&$name`. I will need assistance on this. Thank you.